File: src/main/scala/cub/hackerrank/tutorials/a30DaysOfCode/day08/day08.py
#!/usr/bin/env python3

pb = {}
n = int(input())

for _ in range(n):
    s = input().split(' ', 1)
    pb[s[0]] = s[1]

while True:
    try:
        l = input()
    except EOFError:
        break
    print(''.join([l, '=', pb.get(l)]) if l in pb.keys() else 'Not found')

File: src/main/scala/cub/hackerrank/tutorials/a30DaysOfCode/day08/day08.js
function processData(input) {
  const arr = input.split('\n').reverse()
  const n = parseInt(arr.pop(), 10)
  const pb = new Map()
  for (let i = 0; i < n; i += 1) {
    const s = arr.pop().split(' ')
    pb.set(s[0], s[1])
  }
  while (arr.length > 0) {
    const s = arr.pop()
    // eslint-disable-next-line no-console
    console.log(pb.has(s) ? `${s}=${pb.get(s)}` : 'Not found')
  }
}

process.stdin.resume()
process.stdin.setEncoding('ascii')
let _input = ''
process.stdin.on('data', (input) => {
  _input += input
})

process.stdin.on('end', () => {
  processData(_input)
})
